androidHackathonApp/src/main/java/com/kcc/kmmhackathon/androidHackathonApp/ActionButton.kt

```diff
@@ -0,0 +1,81 @@
+package com.kcc.kmmhackathon.androidHackathonApp
+
+import android.R.attr.button
+import android.annotation.TargetApi
+import android.content.Context
+import android.graphics.drawable.GradientDrawable
+import android.graphics.drawable.LayerDrawable
+import android.os.Build
+import android.text.Spannable
+import android.text.SpannableString
+import android.text.style.BackgroundColorSpan
+import android.util.AttributeSet
+import android.widget.ImageView
+import android.widget.LinearLayout
+import android.widget.TextView
+import androidx.annotation.Nullable
+
+
+class ActionButton : LinearLayout {
+    constructor(context: Context) : super(context) {
+        inflateLayout(context)
+    }
+
+    constructor(context: Context, @Nullable attrs: AttributeSet?) : super(
+            context,
+            attrs,
+    ) {
+        inflateLayout(context)
+    }
+
+    constructor(
+            context: Context,
+            @Nullable attrs: AttributeSet?,
+            defStyleAttr: Int
+    ) : super(context, attrs, defStyleAttr) {
+        inflateLayout(context)
+    }
+
+    @TargetApi(Build.VERSION_CODES.LOLLIPOP)
+    constructor(
+            context: Context,
+            @Nullable attrs: AttributeSet?,
+            defStyleAttr: Int,
+            defStyleRes: Int
+    ) : super(context, attrs, defStyleAttr, defStyleRes) {
+        inflateLayout(context)
+    }
+
+    private fun inflateLayout(context: Context) {
+        inflate(context, R.layout.action_button, this)
+    }
+
+    override fun onFinishInflate() {
+        super.onFinishInflate()
+        val textView = findViewById<TextView>(R.id.textView)
+        makeSpannable(textView)
+    }
+
+    private fun makeSpannable(textView: TextView) {
+        val spannable: Spannable = SpannableString(textView.text)
+        spannable.setSpan(
+                BackgroundColorSpan(-0xffdf),
+                0,
+                textView.text.length,
+                Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
+        )
+        textView.text = spannable
+        invalidate()
+    }
+
+    fun setIcon(resourceId: Int) {
```

Comment:
instead of making a fun for this you might be able to get the reference from the layout ID and set `imageView.getImageResourse` without needing the findViewById, unless the icon is gonna change?